includes/modules/order_total/ot_coupon.php

```diff
@@ -592,7 +606,15 @@ function get_order_total($couponCode)
     $orderTotal = $order->info['total'];
     $products = $_SESSION['cart']->get_products();
     for ($i=0, $n=sizeof($products); $i<$n; $i++) {
-      if (!is_product_valid($products[$i]['id'], $couponCode) || !is_coupon_valid_for_sales($products[$i]['id'], $couponCode)) {
+      $is_product_valid = (is_product_valid($products[$i]['id'], $couponCode) && is_coupon_valid_for_sales($products[$i]['id'], $couponCode));
+      $GLOBALS['zco_notifier']->notify(
+        'NOTIFY_OT_COUPON_PRODUCT_VALIDITY', 
+        array(
+            'is_product_valid' => $is_product_valid, 

```

Comment:
Is there any value here in passing $is_product_valid as a modifiable parameter instead? Just in case the observer has reason to override it? 
Can still pass $i as immutable for lookup/matching purposes.

Reply:
My thought is that the `ot_coupon` notifications are _informative_, rather than _overridable_.  This one, for instance, lets an observer note whether/not the current product qualifies for the current coupon ... and implies that the order's tax calculations will be modified for the current product. 

The coupon-processing is complicated enough, IMO, without another entity declaring that a product qualifies for the coupon outside of its Zen Cart configuration.  That said, I'll make the change if you think it's really needed.